pages/tutorials/3.0/window/events.md

```diff
@@ -7,259 +7,329 @@
     else
         std::cout << "wheel type: unknown" << std::endl;
-    std::cout << "wheel movement: " << event.mouseWheelScroll.delta << std::endl;
-    std::cout << "mouse x: " << event.mouseWheelScroll.x << std::endl;
-    std::cout << "mouse y: " << event.mouseWheelScroll.y << std::endl;
+    std::cout << "wheel movement: " << mouseWheelScrolled->delta << std::endl;
+    std::cout << "mouse x: " << mouseWheelScrolled->position.x << std::endl;
+    std::cout << "mouse y: " << mouseWheelScrolled->position.y << std::endl;
 }
 ```
 
 ## The MouseButtonPressed and MouseButtonReleased events
 
 The `sf::Event::MouseButtonPressed` and `sf::Event::MouseButtonReleased` events are triggered when a mouse button is pressed/released.
-
 SFML supports 5 mouse buttons: left, right, middle (wheel), extra #1 and extra #2 (side buttons).
 
-The member associated with these events is `event.mouseButton`, it contains the code of the pressed/released button, as well as the current position of the mouse cursor.
+The data associated with these events contains the code of the pressed/released button, as well as the current position of the mouse cursor.
 
 ```cpp
-if (event.type == sf::Event::MouseButtonPressed)
+if (const auto* mouseButtonPressed = event->getIf<sf::Event::MouseButtonPressed>())
 {
-    if (event.mouseButton.button == sf::Mouse::Right)
+    if (mouseButtonPressed->button == sf::Mouse::Button::Right)
     {
         std::cout << "the right button was pressed" << std::endl;
-        std::cout << "mouse x: " << event.mouseButton.x << std::endl;
-        std::cout << "mouse y: " << event.mouseButton.y << std::endl;
+        std::cout << "mouse x: " << mouseButtonPressed->position.x << std::endl;
+        std::cout << "mouse y: " << mouseButtonPressed->position.y << std::endl;
     }
 }
 ```
 
 ## The MouseMoved event
 
 The `sf::Event::MouseMoved` event is triggered when the mouse moves within the window.
+This event is triggered even if the window isn't focused.
+However, it is triggered only when the mouse moves within the inner area of the window, not when it moves over the title bar or borders.
+
+The data associated with this event contains the current position of the mouse cursor relative to the window.
+
+```cpp
+if (const auto* mouseMoved = event->getIf<sf::Event::MouseMoved>())
+{
+    std::cout << "new mouse x: " << mouseMoved->position.x << std::endl;
+    std::cout << "new mouse y: " << mouseMoved->position.y << std::endl;
+}
+```
+
+## The MouseMovedRaw event
+
+The `sf::Event::MouseMovedRaw` event is triggered when the mouse moved within the window even if the mouse is moved a distance too small to perceive.
 
-This event is triggered even if the window isn't focused. However, it is triggered only when the mouse moves within the inner area of the window, not when it moves over the title bar or borders.
+While the MouseMoved position value is dependent on the screen resolution, raw data is not.
```

Comment:
```suggestion
While the `sf::Event::MouseMoved` position value is dependent on the screen resolution, the raw data of this event is not.
```

Reply:
I feel like this description isn't enough. We now just know what it is not, but we still don't know, what the data *is*.